batch/batch/server/server.py

```diff
@@ -663,10 +689,12 @@ def run_once(target, *args, **kwargs):
         log.info(f'run_forever: {target.__name__} returned')
     except Exception:  # pylint: disable=W0703
         log.error(f'run_forever: {target.__name__} caught_exception: ', exc_info=sys.exc_info())
+        #  FIXME: Should this raise an exception? Server tests log this, but do not fail
```

Comment:
@danking @cseed See FIXME comment. Tests are in `test_batch_server.py`

Reply:
I don't think so.  If this throws, run_forever exits (it has no try/except), which it never should.

Reply:
I'm seeing this error message that doesn't cause a test failure -- It also occurs on master:

```
test/test_batch_server.py RUNNING
INFO	| 2019-03-11 11:39:45,073 	| server.py 	| run_once:687 | run_forever: urllib3_timeout
ERROR	| 2019-03-11 11:39:45,073 	| server.py 	| run_once:691 | run_forever: urllib3_timeout caught_exception:
Traceback (most recent call last):
  File "/Users/jigold/hail/batch/batch/server/server.py", line 688, in run_once
    target(*args, **kwargs)
  File "/Users/jigold/hail/batch/test/test_batch_server.py", line 17, in urllib3_timeout
    host='127.0.0.1', port=5000), 'url://stuff', err)
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='127.0.0.1', port=5000): urllib3 timeout
INFO	| 2019-03-11 11:39:45,074 	| server.py 	| run_once:687 | run_forever: standard_timeout
ERROR	| 2019-03-11 11:39:45,074 	| server.py 	| run_once:691 | run_forever: standard_timeout caught_exception:
Traceback (most recent call last):
  File "/Users/jigold/hail/batch/batch/server/server.py", line 688, in run_once
    target(*args, **kwargs)
  File "/Users/jigold/hail/batch/test/test_batch_server.py", line 20, in standard_timeout
    raise requests.exceptions.ReadTimeout(err)
requests.exceptions.ReadTimeout: standard timeouut
INFO	| 2019-03-11 11:39:45,074 	| server.py 	| run_once:687 | run_forever: socket_timeout
ERROR	| 2019-03-11 11:39:45,074 	| server.py 	| run_once:691 | run_forever: socket_timeout caught_exception:
Traceback (most recent call last):
  File "/Users/jigold/hail/batch/batch/server/server.py", line 688, in run_once
    target(*args, **kwargs)
  File "/Users/jigold/hail/batch/test/test_batch_server.py", line 23, in socket_timeout
    raise socket.timeout(err)
socket.timeout: socket timeout
```

Reply:
Those errors are being raised by test_batch_server.py; just testing that desired errors are caught.

```py
def test_errors(self):
        def urllib3_timeout(err):
            raise urllib3.exceptions.ReadTimeoutError(HTTPConnectionPool(
                host='127.0.0.1', port=5000), 'url://stuff', err)

        def standard_timeout(err):
            raise requests.exceptions.ReadTimeout(err)

        def socket_timeout(err):
            raise socket.timeout(err)
```

corresponding with lines 17 (`raise urllib3.exceptions.ReadTimeoutError`), 20(`requests.exceptions.ReadTimeout`), etc